logger.c

```diff
@@ -254,27 +254,75 @@ static int _logger_thread_parse_extw(logentry *e, char *scratch) {
 
 static int _logger_thread_parse_cne(logentry *e, char *scratch) {
     int total;
+    unsigned short rport = 0;
+    char rip[64];
+    struct logentry_conn_event *le = (struct logentry_conn_event *) e->data;
     const char * const transport_map[] = { "local", "tcp", "udp" };
 
-    struct logentry_conn_event *le = (struct logentry_conn_event *) e->data;
+    memset(rip, 0, sizeof(rip));
+
+    switch (le->addr.sa_family) {
+        case AF_INET:
+            inet_ntop(AF_INET, &((struct sockaddr_in *) &le->addr)->sin_addr,
+                    rip, sizeof(rip) - 1);
+            rport = ntohs(((struct sockaddr_in *) &le->addr)->sin_port);
+            break;
+        case AF_INET6:
+            inet_ntop(AF_INET6, &((struct sockaddr_in6 *) &le->addr)->sin6_addr,
+                    rip, sizeof(rip) - 1);
+            rport = ntohs(((struct sockaddr_in6 *) &le->addr)->sin6_port);
+            break;
+#ifndef DISABLE_UNIX_SOCKET
+        // Connections on Unix socket transports have c->request_addr zeroed out.
+        case AF_UNSPEC:
+        case AF_UNIX:
+            strncpy(rip, "unix", strlen("unix") + 1);
+            break;
+#endif /* #ifndef DISABLE_UNIX_SOCKET */
+    }
+
     total = snprintf(scratch, LOGGER_PARSE_SCRATCH,
             "ts=%d.%d gid=%llu type=conn_new rip=%s rport=%hu transport=%s cfd=%d\n",
             (int) e->tv.tv_sec, (int) e->tv.tv_usec, (unsigned long long) e->gid,
-            le->rip, le->rport, transport_map[le->transport], le->sfd);
+            rip, rport, transport_map[le->transport], le->sfd);
 
     return total;
 }
 
 static int _logger_thread_parse_cce(logentry *e, char *scratch) {
     int total;
+    unsigned short rport = 0;
+    char rip[64];
+    struct logentry_conn_event *le = (struct logentry_conn_event *) e->data;
     const char * const transport_map[] = { "local", "tcp", "udp" };
     const char * const reason_map[] = { "error", "normal", "idle_timeout", "shutdown" };
 
-    struct logentry_conn_event *le = (struct logentry_conn_event *) e->data;
+    memset(rip, 0, sizeof(rip));
```

Comment:
make a __logger_util_something to avoid the redundancy?

Reply:
Pushed up c722073b7189566e24c688415a4234c1f1a8280d; let me know what you think.

Reply:
Looks good, squash it up and clean up anything else you want?

I need to change some things around in the logger code, so trying to get these in first so I don't invalidate your PR's :)

Reply:
Thanks! I don't think there's anything else I want to change so I've squashed everything down.